Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs

```diff
@@ -174,8 +174,16 @@ private void InitializeServiceLocator()
                 Utilities.Editor.InputMappingAxisUtility.CheckUnityInputManagerMappings(Definitions.Devices.ControllerMappingLibrary.UnityInputManagerAxes);
 #endif
 
-                if (!RegisterService(typeof(IMixedRealityInputSystem), Activator.CreateInstance(ActiveProfile.InputSystemType) as IMixedRealityInputSystem) ||
-                    InputSystem == null)
+                bool registrationSucceeded = false;
+                try
+                {
+                    registrationSucceeded = RegisterService(typeof(IMixedRealityInputSystem), Activator.CreateInstance(ActiveProfile.InputSystemType) as IMixedRealityInputSystem);
```

Comment:
What if we had `RegisterService<T>(Func<T> factory)` and simplified this across all of them?

Reply:
Callers would do `RegisterService<IMixedRealityInputSystem>(() => Activator.CreateInstance(…))`

Reply:
I had thought of that as a possible optimization.